bitrise.yml

```diff
@@ -3,51 +3,221 @@ default_step_lib_source: https://github.com/bitrise-io/bitrise-steplib.git
+      - path::./:
+          inputs:
+            - carthage_options: --platform ios --project-directory $BITRISE_SOURCE_DIR
+
+  test-cache-xcconfig-from-env:
+    before_run:
+      - setup_clean_repo
+      - apply_xcode_workaround
     after_run:
-    - test
+      - assert-outputs
+      - teardown
+    steps:
+      - change-workdir:
+          title: Switch working dir to test / _tmp dir
+          inputs:
+            - path: $ORIG_BITRISE_SOURCE_DIR/_tmp
+            - is_create_path: true
+      - script:
+          inputs:
+            - content: |-
+                #!/bin/bash
+                envman add --key XCODE_XCCONFIG_FILE --value "$TMP_XCODE_XCCONFIG_FILE"
+      - path::./:
+          inputs:
+            - carthage_options: --platform ios --project-directory $BITRISE_SOURCE_DIR
+      - path::./:
+          inputs:
+            - carthage_options: --platform ios --project-directory $BITRISE_SOURCE_DIR
+
+  test-xcconfig-from-input:
+    before_run:
+      - setup_clean_repo
+      - apply_xcode_workaround
+    after_run:
+      - assert-outputs
+      - teardown
+    steps:
+      - change-workdir:
+          title: Switch working dir to test / _tmp dir
+          inputs:
+            - path: $ORIG_BITRISE_SOURCE_DIR/_tmp
+            - is_create_path: true
+      - path::./:
+          inputs:
+            - xcconfig: file://$TMP_XCODE_XCCONFIG_FILE
+            - carthage_options: --platform ios --project-directory $BITRISE_SOURCE_DIR
+
+  test-xcconfig-from-input-url:
+    before_run:
+      - setup_clean_repo
+      - apply_xcode_workaround
+    after_run:
+      - assert-outputs
+      - teardown
+    steps:
+      - change-workdir:
+          title: Switch working dir to test / _tmp dir
+          inputs:
+            - path: $ORIG_BITRISE_SOURCE_DIR/_tmp
+            - is_create_path: true
+      - path::./:
+          inputs:
+            - xcconfig: https://raw.githubusercontent.com/bitrise-io/sample-apps-carthage-objc/master/static.xcconfig
+            - carthage_options: --platform ios --project-directory $BITRISE_SOURCE_DIR
+
+  setup_clean_repo:
+    steps:
+      - change-workdir:
+          title: Switch working dir to ORIG_BITRISE_SOURCE_DIR
+          inputs:
+            - path: $ORIG_BITRISE_SOURCE_DIR
+      - script:
+          title: Clean _tmp folder
+          inputs:
+            - content: |-
+                #!/bin/bash
+                set -e
+                set -v
+                rm -rf $ORIG_BITRISE_SOURCE_DIR/_tmp
+      - script:
+          title: Clear BITRISE_CACHE_INCLUDE_PATHS env
+          inputs:
+            - content: |-
+                #!/bin/bash
+                set -ex
+                envman add --key BITRISE_CACHE_INCLUDE_PATHS --value ""
+      - script:
+          title: Clone sapmle app
+          inputs:
+            - content: git clone $SAMPLE_APP_URL $ORIG_BITRISE_SOURCE_DIR/_tmp
+
+  apply_xcode_workaround:
```

Comment:
Where is this coming from?

Reply:
I've linked the source to line 185, it is coming from a Carthage issue.
`# source https://github.com/Carthage/Carthage/issues/3019#issuecomment-665136323`

Reply:
Oh I didn't notice, looks like I ignored that along with `#!/bin/bash` 🙈 
Btw, did you find this issue during development, or did we always had the intention of fixing this?

Reply:
This workaround is what should be the content of the file at `XCODE_XCCONFIG_FILE`. This workflow just prepares the environment.

Reply:
I'd consider naming this to indicate that here we're building a tmp _local_ `.xcconfig` file which contains the workaround, so stg like `create-local-workaround-xcconfig-file`